src/js/PSVUtils.js

```diff
@@ -683,67 +675,66 @@ PSVUtils.cleanTHREEScene = function(scene) {
  */
 PSVUtils.animation = function(options) {
-  var defer = D(false); // alwaysAsync = false to allow immediate resolution of "cancel"
-  var start = null;
 
-  if (!options.easing || typeof options.easing === 'string') {
-    options.easing = PSVUtils.animation.easings[options.easing || 'linear'];
-  }
+  var promise = new Promise(function(resolve, reject) {
 
-  function run(timestamp) {
-    // the animation has been cancelled
-    if (defer.promise.getStatus() === -1) {
-      return;
-    }
+    var start = null;
 
-    // first iteration
-    if (start === null) {
-      start = timestamp;
+    if (!options.easing || typeof options.easing === 'string') {
+      options.easing = PSVUtils.animation.easings[options.easing || 'linear'];
     }
 
-    // compute progress
-    var progress = (timestamp - start) / options.duration;
-    var current = {};
-    var name;
+    function run(timestamp) {
 
-    if (progress < 1.0) {
-      // interpolate properties
-      for (name in options.properties) {
-        current[name] = options.properties[name].start + (options.properties[name].end - options.properties[name].start) * options.easing(progress);
+      // first iteration
+      if (start === null) {
+        start = timestamp;
       }
 
-      options.onTick(current, progress);
+      // compute progress
+      var progress = (timestamp - start) / options.duration;
+      var current = {};
+      var name;
 
-      window.requestAnimationFrame(run);
-    }
-    else {
-      // call onTick one last time with final values
-      for (name in options.properties) {
-        current[name] = options.properties[name].end;
+      if (progress < 1.0) {
+        // interpolate properties
+        for (name in options.properties) {
+          current[name] = options.properties[name].start + (options.properties[name].end - options.properties[name].start) * options.easing(progress);
+        }
+
+        options.onTick(current, progress);
+
+        window.requestAnimationFrame(run);
       }
+      else {
+        // call onTick one last time with final values
+        for (name in options.properties) {
+          current[name] = options.properties[name].end;
+        }
 
-      options.onTick(current, 1.0);
+        options.onTick(current, 1.0);
 
-      window.requestAnimationFrame(function() {
-        defer.resolve();
-      });
+        window.requestAnimationFrame(function() {
+          resolve();
+        });
+      }
     }
-  }
 
-  if (options.delay !== undefined) {
-    window.setTimeout(function() {
+    if (options.delay !== undefined) {
+      window.setTimeout(function() {
+        window.requestAnimationFrame(run);
+      }, options.delay);
+    }
+    else {
       window.requestAnimationFrame(run);
-    }, options.delay);
-  }
-  else {
-    window.requestAnimationFrame(run);
-  }
+    }
+
+  });
 
-  // add a "cancel" to the promise
-  var promise = defer.promise;
   promise.cancel = function() {
-    defer.reject();
+    reject();
```

Comment:
This won't work, `reject` is not defined here. And anyway as you dropped the status test is does nothing. 

But we need it. This is critical when the move deceleration animation is performed and the user starts a new move.